web/tests/calculate-percentage-directive-spec.js

```diff
@@ -43,13 +43,21 @@ describe('calculatePercentage directive', function() {
             this.elm = this.$compile(tpl)(this.scope);
             this.scope.$digest();
         };
+        spyOn(this.calculateCompletionPercent, 'countQuestionsTotal').and.returnValue(questionSet);
+        spyOn(this.calculateCompletionPercent, 'countNumberOfAnsweredQuestions').and.returnValue(answers);
     });
 
     it('should call this.calculateCompletionPercent.countQuestionsTotal with the questionSet', function() {
-        spyOn(this.calculateCompletionPercent, 'countQuestionsTotal').and.returnValue(questionSet);
+        this.scope.field = questionSet;
+        this.scope.getResponse = getResponse;
+        this.compileDirective(this.tpl);
+        expect(this.calculateCompletionPercent.countQuestionsTotal).toHaveBeenCalled()
     });
 
     it('should call this.calculateCompletionPercent.countNumberOfAnsweredQuestions with the answer set', function() {
-        spyOn(this.calculateCompletionPercent, 'countNumberOfAnsweredQuestions').and.returnValue(answers);
+        this.scope.field = questionSet;
+        this.scope.getResponse = getResponse;
+        this.compileDirective(this.tpl);
+        expect(this.calculateCompletionPercent.countNumberOfAnsweredQuestions).toHaveBeenCalled()
```

Comment:
What I would only change is that you can check that it has been called with the describe values. Use `toHaveBeenCalledWith()`